### PR TITLE
Run Cluster Create/Update/Delete tasks sequentially per SDDC

### PR DESCRIPTION
**Pull request by dimitarproynov**
If two concurrent cluster tasks run per SDDC we get the following error:
com.vmware.skyscraper.common.InternalSkyscraperException: Concurrent update on <SDDC-ID>, column name {1}. Try operation again.

fix for: https://github.com/vmware/terraform-provider-vmc/issues/93

Use a per SDDC lock to force the Terraform provider to wait for the previous task completion.

Now that an almost full E2E suite is present, the pattern of the copy-pasted "wait until finished or error" code across the provider's resources can be extracted, as a helper function to avoid code duplication, safely.

In addition to the refactorings a Spectral scan action has been added following a recommendation from a VMware security architect. 

Testing done:
make build
make test
golangci-lint run

Added a second cluster in the TestAccResourceVmcClusterZerocloud

ran successfully all acceptance tests:
make testacc TESTARGS="-run='TestAccResourceVmcSddcZerocloud|TestAccResourceVmcClusterZerocloud|TestAccResourceVmcSiteRecoveryZerocloud|TestAccResourceVmcSrmNodeZerocloud|TestAccDataSourceVmcCustomerSubnetsBasic|TestAccDataSourceVmcConnectedAccountsBasic|TestAccDataSourceVmcOrgBasic|TestAccDataSourceVmcSddcBasic' -parallel 4

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

### CODE REVIEW

File: vmc/resource_vmc_sddc.go [hunk 1]
```diff
@@ -787,26 +724,14 @@ func resourceSddcUpdate(d *schema.ResourceData, m interface{}) error {
 		publishClient := msft_licensing.NewPublishClient(connectorWrapper)
 		task, err := publishClient.Post(orgID, sddcID, primaryCluster.ClusterId, *configChangeParam)
 		if err != nil {
-			return fmt.Errorf("Error updating license : %s", err)
+			return fmt.Errorf("error updating license : %s", err)
 		}
 		return resource.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-			tasksClient := orgs.NewTasksClient(connectorWrapper)
-			task, err := tasksClient.Get(orgID, task.Id)
-			if err != nil || *task.Status == "FAILED" {
```
Comment (fstrudel) [inline vmc/resource_vmc_sddc.go:795]: nit: microsoft -> Microsoft

File: vmc/sync_utils.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/* Copyright 2022 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package vmc
```
Comment (fstrudel) [inline vmc/sync_utils.go]: As a general Golang rule, avoid `util` and `helper` naming, see https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

Sounds like this file provides tasks related methods?

File: vmc/sync_utils_test.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
+package vmc
```
Comment (fstrudel) [inline vmc/sync_utils_test.go]: always add the proper copyright

File: vmc/sync_utils_test.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	time.Sleep(500 * time.Millisecond)
+	assert.False(t, lock1Obtained)
+	assert.True(t, lock2Obtained)
+	// Test the unlock functionality
+	unlockFunction()
+	time.Sleep(500 * time.Millisecond)
+	assert.True(t, lock1Obtained)
+}
+
+type AuthenticatorMock struct {
```
Comment (fstrudel) [inline vmc/sync_utils_test.go]: Those are not really mocks but stubs. 
You might want to generate the mocks actually to use proper mocks? See https://github.com/golang/mock

File: vmc/sync_utils_test.go (hunk 3 above)
Comment (dimitarproynov) [reply vmc/sync_utils_test.go]: Added copyright headers and updated all the stale ones.

File: vmc/sync_utils_test.go (hunk 4 above)
Comment (dimitarproynov) [reply vmc/sync_utils_test.go]: Renamed to Stubs

File: vmc/sync_utils.go (hunk 2 above)
Comment (dimitarproynov) [reply vmc/sync_utils.go]: Extracted the task related functions to a separate package.

**Review by fstrudel — APPROVED** [review]

**Pull request merged**